domain/resource/state/resource.go

```diff
@@ -522,21 +530,21 @@ WHERE  uuid = $resourceKind.uuid
 	return kind, err
 }
 
-// insertFileResource checks that the storage ID corresponds to stored object
+// upsertFileResource checks that the storage ID corresponds to stored object
 // store metadata and then records that the resource is stored at the provided
 // storage ID.
-func (st *State) insertFileResource(
+func (st *State) upsertFileResource(
```

Comment:
thought: this method isn't really doing an upsert per se, using `ON CONFLICT(uuid) DO UPDATE SET` in the sql, perhaps `replaceFileResource` is a better name.

Reply:
Changed.